### PR TITLE
fix(Model): findOrCreate should fail explicitely, not timeout, when a…

### PR DESCRIPTION
**Pull request by mickhansen**
…ltering hooks before create

### CODE REVIEW

File: lib/model.js [hunk 1]
```diff
@@ -631,12 +631,8 @@ class Model {
       }
     }.bind(this));
 
-    this.attributes = this.rawAttributes = _.mapValues(attributes, function(attribute, name) {
-      if (!Utils._.isPlainObject(attribute)) {
-        attribute = { type: attribute };
-      }
-
-
+    this._fieldToAttributeMap = {};
```
Comment (janmeier) [inline lib/model.js]: remove

File: test/integration/model/create.test.js [hunk 2]
```diff
@@ -394,6 +394,40 @@ describe(Support.getTestDialectTeaser('Model'), function() {
         });
       }
 
+      it.only('should not fail silently with concurrency higher than pool, a unique constraint and a create hook resulting in mismatched values', function() {
```
Comment (janmeier) [inline test/integration/model/create.test.js]: !

File: test/integration/model/create.test.js [hunk 3]
```diff
@@ -394,6 +394,40 @@ describe(Support.getTestDialectTeaser('Model'), function() {
...
+          'mick ',
+          'mick ',
+          'mick '
+        ];
+
+        return User.sync({force: true}).then(() => {
+          return Promise.all(
+            names.map(username => {
+              return User.findOrCreate({where: {username}}).catch(err => {
+                expect(err.message).to.equal(`user#findOrCreate: value used for username was not equal for both the find and the create calls, 'mick ' vs 'mick'`);
```
Comment (janmeier) [inline test/integration/model/create.test.js]: Maybe add a spy here and expect it to have been called?

**Comment by mickhansen** [comment]
Might need/want to hold off untill https://github.com/sequelize/sequelize/pull/6009 is merged

File: lib/dialects/mssql/query.js [hunk 4]
```diff
@@ -210,12 +211,12 @@ Query.prototype.formatError = function (err) {
     if (uniqueKey && !!uniqueKey.msg) {
       message = uniqueKey.msg;
     }
-    if (!!match[2]) {
```
Comment (janmeier) [inline lib/dialects/mssql/query.js]: Maybe the second match is for an old version ?

**Comment by janmeier** [comment]
LGTM

File: lib/dialects/mssql/query.js (hunk 4 above)
Comment (mickhansen) [inline lib/dialects/mssql/query.js]: No idea, not sure if it's ever been correct, but it certainly isn't at the moment.

**Pull request merged**